pkg/dbx/client_test.go

```diff
@@ -0,0 +1,219 @@
+package dbx_test
+
+import (
+	"context"
+	"testing"
+
+	goSqlMock "github.com/DATA-DOG/go-sqlmock"
+	"github.com/jmoiron/sqlx"
+	"github.com/justtrackio/gosoline/pkg/db"
+	"github.com/justtrackio/gosoline/pkg/dbx"
+	"github.com/justtrackio/gosoline/pkg/exec"
+	logMocks "github.com/justtrackio/gosoline/pkg/log/mocks"
+	"github.com/stretchr/testify/suite"
+)
+
+type TestEntity struct {
+	Id      int    `db:"id"`
+	Name    string `db:"name"`
+	Enabled bool   `db:"enabled"`
+}
+
+func TestClientTestSuite(t *testing.T) {
+	suite.Run(t, new(ClientTestSuite))
+}
+
+type ClientTestSuite struct {
+	suite.Suite
+
+	sqlMock goSqlMock.Sqlmock
+	client  dbx.Client[TestEntity]
+}
+
+func (s *ClientTestSuite) SetupTest() {
+	dbMock, sqlMock, err := goSqlMock.New(goSqlMock.QueryMatcherOption(goSqlMock.QueryMatcherEqual))
+	if err != nil {
+		s.Error(err)
+	}
+	loggerMock := logMocks.NewLoggerMock(logMocks.WithMockAll)
```

Comment:
Do you need to assert expecations on the mock on teardownTest? If I remember correctly, that's only done if you pass the testing.T to the mocks